ocaml/lambda/matching.ml

```diff
@@ -2132,8 +2148,12 @@ let get_expr_args_record ~scopes head (arg, _mut, sort, layout) rem =
         | Record_unboxed
         | Record_inlined (_, Variant_unboxed) -> arg, sort, layout
         | Record_float ->
-           (* TODO: could optimise to Alloc_local sometimes *)
-           Lprim (Pfloatfield (lbl.lbl_pos, sem, alloc_heap), [ arg ], loc),
+           let am =
+            match am with
+            | None -> alloc_heap
+            | Some am -> transl_alloc_mode am
```

Comment:
`am` could be `None` if filled-in by `expand_record_head`, which I believe belongs to the `_` in patterns such as `{x;y;_}`, which I think is not important wrt what mode it uses?